src/sqlfluff/api/simple.py

```diff
@@ -0,0 +1,54 @@
+"""The simple public API methods."""
+
+from ..core import Linter
+
+
+def _unfiy_str_or_file(sql):
+    """Unify string and files in the same format."""
+    if not isinstance(sql, str):
+        try:
+            sql = sql.read()
+        except AttributeError:
+            raise TypeError("Value passed as sql is not a string or a readable object.")
+    return sql
+
+
+def lint(sql, dialect="ansi"):
+    """Lint a sql string or file.
+
+    Args:
+        sql (:obj:`str` or file-like object): The sql to be linted
+            either as a string or a subclass of :obj:`TextIOBase`.
+        dialect (:obj:`str`, optional): A reference to the dialect of the sql
+            to be linted. Defaults to `ansi`.
+
+    Returns:
+        :obj:`list` of :obj:`dict` for each violation found.
+    """
+    sql = _unfiy_str_or_file(sql)
+    linter = Linter(dialect=dialect)
+
+    result = linter.lint_string_wrapped(sql)
+    result_records = result.as_records()
+    # Return just the violations for this file
+    return result_records[0]["violations"]
+
+
+def fix(sql, dialect="ansi"):

```

Comment:
Should we consider letting the caller specify which fixes to apply?

Reply:
similar to the rules one, I'll add that in. 👍